design/open/openapi-multipart-upload.md

```diff
@@ -0,0 +1,215 @@
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CompleteMultipartUpload"
+      responses:
+        200:
+          description: Multipart upload completed
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ObjectStats"
+        # ... other error responses
+
+    delete:
+      tags:
+        - staging
+      operationId: abortMultipartUpload
+      summary: Abort a multipart upload
+      description: Aborts a multipart upload.
+      responses:
+        204:
+          description: Multipart upload aborted
+
+```
+
+
+**Schemas used by operations**
+
+```yaml
+components:
+  schemas:
+    MultipartUpload:
+      type: object
+      properties:
+        upload_id:
+          type: string
+        locations:
+          type: array
+          items:
+            $ref: "#/components/schemas/StagingLocation"
+      required:
+        - upload_id
+        
+    UploadPart:
+      type: object
+      properties:
+        part_number:
+          type: integer
+        etag:
+          type: string
+      required:
+        - part_number
+        - etag
+        
+    CompleteMultipartUpload:
+      type: object
+      properties:
+        parts:
+          type: array
+          description: "List of uploaded parts, should be ordered by ascending part number"
+          items:
+            $ref: "#/components/schemas/UploadPart"
+      required:
+        - parts
+```
+
+### Support and discover
+
+Presign support is a capability lakectl discover before switching to use presign for upload or download from lakeFS.
+The multipart upload support will be part of the storage capability add `pre_sign_support_parts` optional field that when set to `true` the user can perform multipart upload using the new API.
+
+```yaml
+    StorageConfig:
+      type: object
+      required:
+        - blockstore_type
+        - blockstore_namespace_example
+        - blockstore_namespace_ValidityRegex
+        - pre_sign_support
+        - pre_sign_support_ui
+        - import_support
+        - import_validity_regex
+      properties:
+        blockstore_type:
+          type: string
+        blockstore_namespace_example:
+          type: string
+        blockstore_namespace_ValidityRegex:
+          type: string
+        default_namespace_prefix:
+          type: string
+        pre_sign_support:
+          type: boolean
+        pre_sign_support_ui:
+          type: boolean
+        import_support:
+          type: boolean
+        import_validity_regex:
+          type: string
+        pre_sign_support_parts:
+          type: boolean
+```
+
+## Limitations
+
+- **S3 block adapter exclusive:** Uploading files in multiple parts (multipart upload) is only available when using the S3 block adapter. This feature isn't currently supported with other storage options.
```

Comment:
Maybe mention here that multipart uploads are available through the S3 gateway to other block adapters so readers won't get the wrong idea that multipart uploads aren't supported at all for non-S3 users.